updater.sh

```diff
@@ -111,18 +112,21 @@ finish_up() {
         nohup bash -c "sleep 5 ; reboot " > /dev/null 2>&1 &
     fi
     sleep 2
-    kill $tail_pid || true
+    if [[ -n ${tail_pid+x} ]]; then
```

Comment:
I'm unsure of the logic in this stanza, and in particular what the `${tail_pid+x}` is doing.  Can you explain it a bit?